sycl/include/CL/sycl/detail/aligned_allocator.hpp

```diff
@@ -56,7 +56,7 @@ template <typename T> class aligned_allocator {
 
     pointer Result = reinterpret_cast<pointer>(
         detail::OSUtil::alignedAlloc(MAlignment, NumBytes));
-    if (!Result)
+    if (!Result || NumBytes == 0)
```

Comment:
I think the beginning of the function is more appropriate place for this check i.e. argument value check.
```c++
pointer allocate(size_t Size) {
  if (0 == Size)
    throw std::bad_alloc();
  size_t NumBytes = ...
```